org.eclipse.xtext.tests/src/org/eclipse/xtext/util/MergeableManifest2Test.java

```diff
@@ -1028,6 +1104,50 @@ public void readWrite_realisticManifest() throws Exception {
 		assertEquals(content, write(manifest));
 	}
 
+	@Test
+	public void readWrite_addRequiredBundles_realisticManifest_regression() throws Exception {
+		// @formatter:off
+        String content =
+            "Manifest-Version: 1.0" + NL + 
+            "Bundle-ManifestVersion: 2" + NL + 
+            "Bundle-Name: Xtext Homeautomation Example - Runtime" + NL + 
+            "Bundle-Vendor: Eclipse Xtext" + NL + 
+            "Bundle-Version: 2.16.0.qualifier" + NL + 
+            "Bundle-SymbolicName: org.eclipse.xtext.example.homeautomation; singleton:=true" + NL + 
+            "Bundle-ActivationPolicy: lazy" + NL + 
+            "Require-Bundle: org.eclipse.xtext," + NL + 
+            " org.eclipse.xtext.xbase," + NL + 
+            " org.eclipse.equinox.common;bundle-version=\"3.5.0\"," + NL + 
+            " org.eclipse.emf.ecore," + NL + 
+            " org.objectweb.asm;bundle-version=\"[6.2.1,6.3.0)\";resolution:=optional," + NL + 
+            " org.eclipse.xtext.util," + NL + 
+            " org.antlr.runtime;bundle-version=\"[3.2.0,3.2.1)\"," + NL + 
+            " org.eclipse.xtend.lib;bundle-version=\"2.14.0\"," + NL + 
+            " org.eclipse.emf.common," + NL + 
+            " org.eclipse.xtext.common.types;bundle-version=\"2.14.0\";visibility:=reexport," + NL + 
```

Comment:
Please add a test where bundle-version and visibility are present in a different order

Reply:
Added such a test.